optaplanner-examples/src/main/java/org/optaplanner/examples/vehiclerouting/domain/location/segmented/HubSegmentLocation.java

```diff
@@ -21,10 +21,16 @@ public class HubSegmentLocation extends Location {
     public HubSegmentLocation() {
     }
 
+    public HubSegmentLocation(long id) {
+        super(id);
+    }
+
     public HubSegmentLocation(long id, double latitude, double longitude) {
         super(id, latitude, longitude);
     }
 
+    @JsonSerialize(keyUsing = HubSegmentLocationKeySerializer.class)
+    @JsonDeserialize(keyUsing = HubSegmentLocationKeyDeserializer.class)
     public Map<RoadSegmentLocation, Double> getNearbyTravelDistanceMap() {
```

Comment:
Confused: road segment being deserialized with hub segment deserializer?

Reply:
Good catch; I am about to provide a unit test before I un-draft the PR. This example confirms we need one, as the segmented locations are not used in any JSON dataset.